cmd/kubeadm/app/constants/constants_test.go

```diff
@@ -198,6 +198,61 @@ func TestEtcdSupportedVersion(t *testing.T) {
 	}
 }
 
+func TestEtcdNearestVersion(t *testing.T) {
+	var tests = []struct {
+		kubernetesVersion string
+		expectedVersion   *version.Version
+		expectedError     error
+	}{
+		{
+			kubernetesVersion: "1.10.0",
+			expectedVersion:   version.MustParseSemantic("3.2.24"),
```

Comment:
sadly, this will fail once we update the etcd struct.
the idea is to not make unit tests depend on version pinning.

one solution here is to make EtcdNearestVersion accept a source `map[uint8]string` map.